mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java

```diff
@@ -44,11 +46,20 @@
     public final class AddedImpl extends MailboxListener.Added implements MailboxAware {
         private final Map<MessageUid, MessageMetaData> added;
         private final Mailbox mailbox;
+        private final Optional<MailboxMessage> message;
 
         public AddedImpl(MailboxSession session, Mailbox mailbox, SortedMap<MessageUid, MessageMetaData> uids) {
             super(session, new StoreMailboxPath(mailbox));
             this.added = ImmutableMap.copyOf(uids);
             this.mailbox = mailbox;
+            this.message = Optional.absent();
+        }
+
+        public AddedImpl(MailboxSession session, Mailbox mailbox, MailboxMessage message, MessageMetaData messageMetaData) {
```

Comment:
I hesitate between this modification and introducing a new kind of event: _SingleMessageAdded_